osu.Game.Rulesets.Osu/Mods/OsuModArrange.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Graphics;
+using osu.Game.Graphics;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects;
+using OpenTK;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    internal class OsuModArrange : Mod, IApplicableToDrawableHitObjects
+    {
+        public override string Name => "Arrange";
+        public override string ShortenedName => "Arrange";
+        public override FontAwesome Icon => FontAwesome.fa_arrows;
+        public override ModType Type => ModType.Fun;
+        public override string Description => "Everything rotates. EVERYTHING";
+        public override double ScoreMultiplier => 1.05;
+
+        public void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables)
+        {
+            drawables.ForEach(drawable => drawable.ApplyCustomUpdateState += drawableOnApplyCustomUpdateState);
+        }
+
+        private float theta;
+
+        private void drawableOnApplyCustomUpdateState(DrawableHitObject drawable, ArmedState state)
+        {
+            var hitObject = (OsuHitObject) drawable.HitObject;
+
+            // repeat points get their position data from the slider.
+            if (hitObject is RepeatPoint)
+                return;
+
+            Vector2 originalPosition = drawable.Position;
+
+            // avoiding that the player can see the abroupt move.
+            const int pre_time_offset = 1000;
+
+            using (drawable.BeginAbsoluteSequence(hitObject.StartTime - hitObject.TimeFadeIn - pre_time_offset, true))
```

Comment:
might be able to use `hitObject.StartTime - hitObject.PreEmpt` here.

Reply:
Sadly no. preempt is not early enough. The hit explosion appears in the wrong location if you use preempt.

Reply:
Have you tried taking one extra millisecond to see if that helps? sounds like a bigger issue if not (which should be figured out, rather than adding an arbitrary number).

Reply:
It works when you add one one extra millisecond.

Reply:
https://drive.google.com/uc?id=1oSaVg-SuQ1RKObCPCADUoiZRJrR2ZL-r
^